### DISCUSSION TITLE
Pedido de opinião sobre o futuro da profissão.

### DISCUSSION THREAD

**Discussion by mauvieira** [frontendbr/forum#1617] [Discussão]
Opa galera, tudo bem?

Meu nome é Mauricio, sou Desenvolvedor Front-end a mais ou menos dois anos (apaixonado por javascript), e de uns tempos pra cá, me peguei tendo alguns questionamentos sobre o futuro da nossa profissão.

Isso começou em uma [palestra da Carla Veira na BrazilJS Conf](https://speakerdeck.com/carlaprv/o-front-end-como-voce-conhece-vai-morrer). Onde ela diz que a profissão vai mudar muito e todo o trabalho **operacional** do Front vai acabar morrendo, tornando-se algo muito mais **pensado e analítico**.

Quando ouvi isso, lembrei na hora de um trecho [em um post do Felipe](https://www.felipefialho.com/blog/o-inicio-o-fim-e-o-meio-do-desenvolvimento-front-end/) que ele diz assim:

> Developers, que sabem HTML, CSS e um pouco de JavaScript terão que participar ativamente no desenvolvimento de aplicações e isso inclui melhorar suas skills como engenheiro de software, ou mesmo migrar para áreas relacionadas a UX e Design.

Acabei caindo em um limbo de questionamentos sobre a profissão em si... Fico com medo de estar estudando e praticando algo que não vai me trazer frutos no futuro, e penso em talvez mudar pra outra área enquanto ainda sou jovem (21 anos), mesmo amando tanto a área de desenvolvimento :/

Já chegaram a pensar algo sobre isso? Acham que desenvolvimento (tanto Back como Front) vai continuar forte no mercado daqui a algum tempo? Ou vale mais a pena migrar pra Data Science, UX ou Machine Learning?

Desde já muito obrigado! espero ter sido claro nesta dúvida/desabafo.

**Comment by talitaazevedo** [comment]
Oie,
Bom dia! 
@mauriciohike Eu sou iniciante, mas  acho que ainda terão uns bons anos  de desenvolvimento front pela frente, a verdade é que o front virou uma coisa complexa tão complexa quanto o backend ou até mais as vezes, as interfaces estão cada vez mais afiadas,   eu vi uma tendência de full stack developer , e também um deficit imenso de profissionais nível pleno e senior dentro da tecnologia principalmente em programação. Eu acho que se a pessoa se qualificar para vagas, ainda vai ter um bom caminho a trilhar nesse quesito de desenvolvimento.

**Comment by tcelestino** [comment]
Já ouvi essa história de que o desenvolvimento front-end "vai morrer" há muito tempo atrás e te digo que na verdade o que vai acontecer é uma evolução da área como acontece desde 2005, com surgimentos de tecnologias, ferramentas, etc...

Se olharmos o mercado de tecnologia em geral, toda a cadeia em volta. E isso vai continuar por muito tempo, já que mesmo com avanço das tecnologias ligadas a inteligência artificial (acho que foi até o foco da palestra da Viera) ainda exigirá a operação do ser humano para lidar como questões relacionados principalmente a "humanização das interfaces".

Sobre o parágrafo do artigo do @felipefialho, olhando para trás, pessoas que lidavam com interfaces deveriam saber sobre design (ux, writer),acessibilidade, usabilidade... isso sempre foi a premissa do front-end e isso vem mudando, acredito eu, pela a forma que vem sendo divulgado o mercado de front-end.

Continue seus estudos, busque bastante conhecimento e se foque no que você acredita, seja front-end, seja ciência de dados, machine learning...

**Comment by mauvieira** [comment]
Obrigado pelas respostas!

Acredito que eu tenha me expressado mal, o meu medo foi sobre o mercado de Desenvolvimento em geral, não só o Front.
Mas é entendo o que vocês disseram, a área vai evoluindo e mudando conforme o tempo, o jeito é não perder o ritmo e se manter na dança enquanto a música rola...

Abraços =)

**Comment by felipefialho** [comment]
Fala @mauriciohike, tudo bem?

Primeiramente desculpa se meu artigo te deixou de alguma forma preocupado, na realidade, a ideia dele era dar uma visão mais otimista do futuro da profissão, aparentemente falhei, hahaha.

Segundo, desconfie de qualquer pessoa que tenta prever o futuro, inclusive e especialmente de mim, a grande verdade é que ninguém consegue prever nada e se acertar é pura cagada. Ninguém poderia prever o lançamento do iPhone em 2007 e muito menos o tanto que isso afetaria a evolução da nossa profissão, com a queda do Flash e acensão do HTML5.

Sendo assim, quando tentamos falar do futuro são achismos com base nas informações que temos agora.

O Front-end teve um destaque enorme nessa década, especialmente após o NodeJS e o ES2015, acredito que nos próximos anos ainda não vamos ter grandes mudanças no cenário atual, mas de novo, isso é com base nas informações que temos agora, e uma opinião minha.

Então o que podemos fazer para nos preparar para mudanças que podem fatalmente acontecer? Aumentar cada vez mais nossa capacidade de adaptação e flexibilidade., como diria Nassim Taleb: 

> Se tornar Antifragil

Apesar desse artigo ser de 2016 (3 anos é bastante coisa), o que eu quis dizer com esse paragrafo é justamente melhorar nossas skills em áreas que tendem a ser mais adaptáveis no mercado: Design e Engenharia de Software.

Espero que esse textão ajude nas suas dúvidas!

_não fecha a issue, o assunto é sempre interessante_

**Comment by eliseumds** [comment]
Nossa área é abrangente e complexa o suficiente pra ter espaço pra todos, por muitos anos. Eu comecei como front-end e hoje faço de tudo um pouco, mas vai de cada um. Evoluções que vi em dez anos de profissão:

- Há menos language wars hoje em dia. A galera entendeu que o que importa são fundamentos, comunidade e especialidade do time
- Designers metendo a mão em código, principalmente React e CSS
- Menores recursos aplicados à infraestrutura. Apesar de ser uma área gigantesca, empresas menores hoje em dia conseguem entregar um produto escalável, seguro, rápido e barato
- As coisas hoje em dia estão menos fragmentadas, eu acho. Docker, Unix, React, Webpack, Slack, Stripe, GSuite e AWS, por ex, têm market shares monstruosos
- O mercado hoje é global, e isso mudou as dinâmicas sociais nas empresas, que procuram trabalhadores fora do país, ou têm que importar

O papo da transição de operacional pra analítico está em todo lugar, e faz sentido. À medida que as ferramentas e processos se tornam mais consolidados, a gente começa a focar mais em regra de negócio. Bom pra todo mundo.

**Comment by iaurg** [comment]
Eu creio que esse trecho do Maquiavel pode lhe dar uma ideia: "Um sábio deve observar as regras e nunca ficar ocioso nos tempos de paz, mas, sim, aumentar os seus recursos com habilidade, de tal forma que estarão à sua disposição na adversidade, a fim de que, quando a sorte mudar, ele esteja preparado para resolver seus problemas."

Um livro de 1513 e já havia essa preocupação, seguindo a ideia do @felipefialho, sempre desconfie de quem prevê o futuro, tudo é incerto.

E seguindo a lógica do trecho citado acima, se você estiver sempre adquirindo conhecimento, participando ativamente de comunidades, vendo o que está rolando de novo, novas tendências, tecnologias, paradigmas e afins, você nem perceberá essa transição.

Minha experiência própria: Eu estava fazendo muita grana com o clássico site institucional, Wordpress e temas, é um mercado que ainda existe mas a tendência é ficar tão simples que os profissionais ficam mais baratos e a concorrência monstruosa, eu estava em um momento de paz e precisei acordar.

Estudo diariamente, principalmente a base de ciência da computação, independente de linguagem, framework, biblioteca e afins a base é a mesma (exceto caso descubram uma nova tecnologia que mude tudo), obviamente foco em aprender e treinar com Javascript que é o forte do mercado hoje, porém se amanhã uma outra linguagem ganhar posição provavelmente a base será similar, precisarei apenas me adaptar a nova sintaxe e alguns padrões.

E por fim faça o que você acreditar que é o correto, evite deixar que os outros guiem o que você deve fazer. A área de tecnologia sempre terá uma porta para você trabalhar, creio que com isso pode ficar tranquilo, o mundo se move com tecnologia.

Recomendo você dar uma olhada nesse vídeo, ele resume como o mercado muda:
[A Dimensão do TEMPO para Programadores](https://www.youtube.com/watch?v=Qb5b8ZE9tIY)

**Comment by ninetails** [comment]
O que vou falar são de opiniões pessoais.

O que eu acho é que você não deve ficar com medo de mudanças, elas sempre vão acontecer. Se é pra melhor ou pra pior é relativo. Um exemplo já citado na _thread_ foi o Flash, quem imaginaria que o HTML5 ia reinar hoje naquela época em que Flash era tudo? E quanta gente perdeu emprego por não se adaptar?

Da mesma forma que hj vc pode pensar que HTML também não pode ser tudo. Só que não é pra ter medo, pelo contrário, encare e acompanhe as tendências. Como citaram pensadores antigos, um ensinamento que cai como uma luva é do Heráclito:

> Ninguém pode entrar duas vezes no mesmo rio, pois quando nele se entra novamente, não se encontra as mesmas águas, e o próprio ser já se modificou.

É muito mais fácil acompanhar e mudar junto com a tecnologia, diluir isso durante sua vida profissional do que fazer escadinhas e sempre ter que aprender um volume muito maior de informação em menos tempo.

Frontend hoje não é o mesmo frontend da época que a gente era chamado de webmasters. Idem backend. Idem tecnologias. Idem consumidores, que hj acessam muito mais em celulares que no Desktop.

Só que não acho que haverão mudanças drásticas tão cedo. Realidade aumentada ou virtual ainda não é realidade, pessoas ainda verão interfaces web utilizando um webview. Estamos numa fase que ainda se faz necessário ter visualização numa tela. Inteligência artificial ficou menos custoso somente agora. Logo ainda acho que o nosso mercado só tem a crescer dado que muitos processos manuais continuam em nossa base: nossa conta de água e luz não vem no APP mostrando o quanto a gente consumiu; ainda temos que assinar contratos e reconhecer firma em cartórios; contratos assinados em papel em todos os lugares.... e por aí vai. Tecnologia ainda precisa muito de frontend no momento. Pense no agora e faça o agora mas esteja preparado para o amanhã. Amanhã pode ser que frontend atual deixa de ser trend/necessidade e tenhamos que fazer mobile/apps nativos (que não deixa de ser frontend).

**Não seja você um software substituível**, aquele que jogamos fora quando não dá mais pra dar manutenção e que precisamos fazer do zero.
